design/FY2019/NFP-AFNFanModel.md

```diff
@@ -0,0 +1,124 @@
+Support System Fan Object in Airflow Networks
+================
+
+**Lixing Gu**
+
+**Florida Solar Energy Center**
+
+ - Original version
+ - 3/11/19 
+
+ 
+
+## Justification for New Feature ##
+
+A new fan object model was recently added to EnergyPlus which will become the only fan object type, eliminating and simplifying the various other options, as it is capable of simulating the same physics and controls as all of the other objects. Currently, the fan is available in most parent model objects in EnergyPlus, but it is not available in airflow network simulations. Since this is supposed to be the new primary fan object, it must be also available in airflow network simulations.
+
+## E-mail and  Conference Call Conclusions ##
+
+NA
+
+## Overview ##
+
+The Fan:SystemModel object models fans of various types using a relatively simple engineering model. This fan can be used in variable air volume, constant volume, on-off cycling, two-speed, or multi-speed applications. It was designed as a replacement for Fan:ConstantVolume, Fan:OnOff, Fan:VariableVolume, and FanPerformance:NightVentilation. In order to make the airflow network model accept the Fan:SystemModel object, an additional choice of fan type input and internal code revision are needed as follows.
+
+- Additional input choice
+
+A new choice key will be added in the Supply Fan Object Type field of the AirflowNetwork:Distribution:Component:Fan object as Fan:SystemModel.
+ 
+- Code revision
+
+Two modules will be revised to accommodate the new feature: AirflowNetworkBalanceManager and HVACFan.  
+ 
+
+## Approach ##
+
+The following revisions will be performed.  Any new additions will be highlighted in red.
+
+### AirflowNetworkBalanceManager
+
+GetAirflowNetworkInput
+
+The function will be modified to accept a new choice of fan type as Fan:SystemModel. Then the code will check the input choice of the Speed Control Method field in the Fan:SystemModel object. If the input is Continuous, the AFN model will consider this fan is a VAV fan. If the input is Discrete, the AFN model will consider this fan is on/off fan. the difference between on/off and constant volume fan is based the value of LoopOnOffFanPartLoadRatio.   
+
+### HAVCFan
+
+#### Assing fan part load value
+The value of OnOffFanPartLoadFraction will be assigned in the update function.
+
+#### Create two local functions to get node numbers of inlet and outlet
+Two local functions of GetFanInletNodeNum and GetFanOutletNodeNum will be created to output node numbers. The functions will be called from GetAirflowNetworkInput.
```

Comment:
Will these two functions be located with the fans in their namespace or will they be in the AFN namespace?

Reply:
@jasondegraw These two functions will be created in the HVACFan module.